pkg/cloudproduct/cloudproduct.go

```diff
@@ -17,38 +17,42 @@ package cloudproduct
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	agonesv1 "agones.dev/agones/pkg/apis/agones/v1"
 	"agones.dev/agones/pkg/client/informers/externalversions"
 	"agones.dev/agones/pkg/cloudproduct/generic"
 	"agones.dev/agones/pkg/cloudproduct/gke"
 	"agones.dev/agones/pkg/portallocator"
 	"agones.dev/agones/pkg/util/runtime"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 )
 
-// CloudProduct provides a generic interface that abstracts cloud product
-// specific functionality. Users should call New() to instantiate a
-// specific cloud product interface.
-type CloudProduct interface {
+// The Cloud Product abstraction currently consists of disjoint interfaces:
+// * ControllerHooks() is an interface available to the pkg/controller binaries
+// * api hooks are separated into api group and registered directly with the API
+//
+// Users should bind flags with CloudProductBind{Flags,Env}, then call
+// InitializeCloudProduct with a client to initialize.
+
+// CloudProductControllerHooks provides a generic interface that abstracts cloud product
+// specific functionality for pkg/controller packages.
+type CloudProductControllerHooks interface {
```

Comment:
```suggestion
type ControllerHooks interface {
```

Seems like it's redundant to have the same name in the interface as we have in the package name
(And repeat through the package to remove the package name from functions and types, etc)

For example: `cloudproduct.CloudProductBindFlags()` could be `cloudproduct.BindFlags()`

Reply:
You're right, fixing.